xtask/src/main.rs

```diff
@@ -96,3 +245,29 @@ fn extract_changelog_for_newest_version(changelog: &str) -> String {
     // The GH API expects those special characters to be replaced.
     release_text.to_string()
 }
+
+fn changelog_header(mut writer: impl std::io::Write, version: &str) -> Result<(), std::io::Error> {
+    writeln!(writer, "## [{}]", version)?;
+    writeln!(writer, "")?;
+    writeln!(writer, "Released {}", chrono::Utc::now().format("%Y-%m-%d"))?;
+    writeln!(writer, "")?;
+
+    Ok(())
+}
+
+fn write_changelog_section(
+    mut writer: impl std::io::Write,
+    heading: &str,
+    fragments: &[PathBuf],
+) -> Result<(), std::io::Error> {
+    writeln!(writer, "### {}", heading)?;
+    writeln!(writer, "")?;
+
+    for fragment in fragments {
+        writeln!(writer, "{}", std::fs::read_to_string(fragment)?.trim_end())?;
```

Comment:
Wouldn't it be nice to make a list here with `-` prefixed without requiring authors to put that bit in the fragment file?

On the same note, we could automatically extract the PR number and the author info from the GH API. But that one we can do in a separate PR.